feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go

```diff
@@ -0,0 +1,179 @@
+
+		if !component.MfgName().Lookup(t).IsPresent() {
+			t.Logf("component.MfgName().Lookup(t).IsPresent() for %q is false. skip it", transceiver)
+		} else {
+			mfgName := component.MfgName().Get(t)
+			t.Logf("Transceiver %s MfgName: %s", transceiver, mfgName)
+
+			//TODO: Remove the mfgName INNOLIGHT check after the issue is fixed.
+			if mfgName == "INNOLIGHT" {
+				t.Logf("Optics from INNOLIGHT is not supported, skip it for now.")
+				continue
+			}
+			inputPowers := component.Transceiver().ChannelAny().InputPower().Instant().Get(t)
+			t.Logf("Transceiver %s inputPowers: %v", transceiver, inputPowers)
+			if len(inputPowers) == 0 {
+				t.Errorf("Get inputPowers list for %q: got 0, want > 0", transceiver)
+			}
+			outputPowers := component.Transceiver().ChannelAny().OutputPower().Instant().Get(t)
+			t.Logf("Transceiver %s outputPowers: %v", transceiver, outputPowers)
+			if len(outputPowers) == 0 {
+				t.Errorf("Get outputPowers list for %q: got 0, want > 0", transceiver)
+			}
+
+			biasCurrents := component.Transceiver().ChannelAny().LaserBiasCurrent().Instant().Get(t)
+			t.Logf("Transceiver %s biasCurrents: %v", transceiver, biasCurrents)
+			if len(outputPowers) == 0 {
+				t.Errorf("Get biasCurrents list for %q: got 0, want > 0", transceiver)
+			}
+		}
+	}
+}
+
+func TestOpticsPowerUpdate(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dp := dut.Port(t, "port1")
+	d := &telemetry.Device{}
+	i := d.GetOrCreateInterface(dp.Name())
+
+	cases := []struct {
+		desc                string
+		IntfStatus          bool
+		expectedStatus      telemetry.E_Interface_OperStatus
+		expectedMaxOutPower float64
+		checkMinOutPower    bool
+	}{{
+		// Check both input and output optics power are in normal range.
+		desc:                "Check initial input and output optics powers are OK",
+		IntfStatus:          true,
+		expectedStatus:      telemetry.Interface_OperStatus_UP,
+		expectedMaxOutPower: maxOpticsPower,
+		checkMinOutPower:    true,
+	}, {
+		desc:                "Check output optics power is very small after interface is disabled",
+		IntfStatus:          false,
+		expectedStatus:      telemetry.Interface_OperStatus_DOWN,
+		expectedMaxOutPower: minOpticsPower,
+		checkMinOutPower:    false,
+	}, {
+		desc:                "Check output optics power is normal after interface is re-enabled",
+		IntfStatus:          true,
+		expectedStatus:      telemetry.Interface_OperStatus_UP,
+		expectedMaxOutPower: maxOpticsPower,
+		checkMinOutPower:    true,
+	}}
+	for _, tc := range cases {
+		t.Log(tc.desc)
+		intUpdateTime := 2 * time.Minute
+		t.Run(tc.desc, func(t *testing.T) {
+			i.Enabled = ygot.Bool(tc.IntfStatus)
+			dut.Config().Interface(dp.Name()).Replace(t, i)
+			dut.Telemetry().Interface(dp.Name()).OperStatus().Await(t, intUpdateTime, tc.expectedStatus)
+
+			// TODO: Remove t.Skipf() after the issue is fixed.
+			t.Skipf("Optics from INNOLIGHT is not supported, skip it for now.")
```

Comment:
Can we make this a conditional skip that detects the manufacturer?

Reply:
done